src/plugins/intel_cpu/src/node.cpp

```diff
@@ -621,44 +624,48 @@ void Node::initSupportedPrimitiveDescriptors() {
+    auto addSupportedPrimitiveDescriptor = [&](const dnnl::primitive_desc& prim_desc) {
+        std::vector<PortConfig> inConfs, outConfs;
+        const int inPlaceOutPort = canBeInPlace() ? 0 : -1;
 
-    for (auto& desc : descs) {
-        primitive_desc_iterator itpd = desc;
-
-        while (static_cast<bool>(itpd)) {
-            NodeConfig config;
-            for (size_t i = 0; i < descInputNumbers(); i++) {
-                PortConfig portConfig;
-                portConfig.inPlace(-1);
-                portConfig.constant(false);
-                auto desc = getSrcMemDesc(itpd, i);
-                if (desc->getType() & MemoryDescType::Blocked) {
-                    portConfig.setMemDesc(std::dynamic_pointer_cast<BlockedMemoryDesc>(desc), BLOCKED_DESC_EMPTY_MASK);
-                } else {
-                    portConfig.setMemDesc(std::move(desc));
-                }
-                config.inConfs.push_back(portConfig);
-            }
+        for (size_t i = 0; i < descInputNumbers(); i++) {
+            auto desc = getSrcMemDesc(prim_desc, i);
 
-            for (size_t i = 0; i < descOutputNumbers(); i++) {
-                PortConfig portConfig;
-                portConfig.inPlace(canBeInPlace() ? 0 : -1);
-                portConfig.constant(false);
-                auto desc = getDstMemDesc(itpd, i);
-                if (desc->getType() & MemoryDescType::Blocked) {
-                    portConfig.setMemDesc(std::dynamic_pointer_cast<BlockedMemoryDesc>(desc), BLOCKED_DESC_EMPTY_MASK);
-                } else {
-                    portConfig.setMemDesc(std::move(desc));
-                }
-                config.outConfs.push_back(portConfig);
-            }
-            impl_desc_type impl_type = parse_impl_name(itpd.impl_info_str());
+            inConfs.emplace_back(desc, BlockedMemoryDesc::BLOCKED_DESC_EMPTY_MASK);
+        }
 
-            supportedPrimitiveDescriptors.emplace_back(config, impl_type);
-            if (!itpd.next_impl())
-                break;
+        for (size_t i = 0; i < descOutputNumbers(); i++) {
+            auto desc = getDstMemDesc(prim_desc, i);
+
+            outConfs.emplace_back(desc, BlockedMemoryDesc::BLOCKED_DESC_EMPTY_MASK, inPlaceOutPort);
         }
+
+        const NodeConfig config(inConfs, outConfs);
+        const impl_desc_type impl_type = parse_impl_name(prim_desc.impl_info_str());
+
+        supportedPrimitiveDescriptors.emplace_back(config, impl_type);
+    };
+
+    for (auto& desc : descs) {
+        auto first_desc = dnnl::primitive_desc(DnnlExtensionUtils::clone_primitive_desc(desc.get()));
+        /* When custom implementation priorities are NOT defined it is enough to
+         * just use the first implementation from priority list.
+         * When custom implementation priorities are defined, all the implementations should be considered,
+         * since custom implementation priorities can be not available at all, so fallback to the default ones must happen */
```

Comment:
Don't actually remember if we discussed this or not. 
Behavior in case custom priority list is provided and not remains different. Is it possible to align it? Like we first go through the list of custom priorities creating primitive desc for each of them and only after that call the logic below for default list?

Reply:
The problem here is that oneDNN primitive_desc is mutating while iterating.
To achieve what you have described we will have to create another copy to iterate over default impl list.

Reply:
Thanks, I would propose to explicitly state this in the comment.

Reply:
done